prism-agent/service/server/src/main/scala/io/iohk/atala/verification/controller/http/VcVerificationParameter.scala

```diff
@@ -0,0 +1,36 @@
+package io.iohk.atala.verification.controller.http
+
+import io.iohk.atala.pollux.core.service.verification.{
+  AudienceParameter as ServiceAudienceParameter,
+  VcVerificationParameter as ServiceVcVerificationParameter
+}
+import sttp.tapir.Schema
+import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}
+
+sealed trait VcVerificationParameter
+
+object VcVerificationParameter {
+  given encoder: JsonEncoder[VcVerificationParameter] =
+    DeriveJsonEncoder.gen[VcVerificationParameter]
+
+  given decoder: JsonDecoder[VcVerificationParameter] =
+    DeriveJsonDecoder.gen[VcVerificationParameter]
+
+  given schema: Schema[VcVerificationParameter] = Schema.derived
+
+  def convert(parameter: VcVerificationParameter): ServiceVcVerificationParameter = {
+    parameter match
+      case AudienceParameter(aud) => ServiceAudienceParameter(aud)
+  }
+}
+
+case class AudienceParameter(aud: String) extends VcVerificationParameter
```

Comment:
Do we really want to keep both the `enum` part and the related `xxxParameter(xxx)` part? It seems like there is always a 1:1 mapping between them. Keeping both allows the user to mix unrelated types together (e.g. `SubjectCheck` with `AudienceParameter`). Why don't we just have `AudienceCheck(aud: String)` instead?

Reply:
You are right, will fix it in follow up